pkg/query/querier.go

```diff
@@ -83,6 +98,11 @@ func (q *querier) Select(ms ...*promlabels.Matcher) storage.SeriesSet {
 		return promSeriesSet{set: errSeriesSet{err: err}}
 	}
 	for _, s := range q.stores {
+		// We might be able to skip the store if its meta information indicates
+		// it cannot have series matching our query.
+		if !storeMatches(s, ms...) {
```

Comment:
This is great, but how does it work for store nodes... ;p

Reply:
Yup, we need similar logic + time based filtering for store nodes.